src/utils/getFiltredPeople.ts

```diff
@@ -0,0 +1,64 @@
+import { Person } from '../types';
+
+type Filters = {
+  sex: string | null;
+  query: string;
+  centuries: string[];
+  sort: string | null;
+  order: string | null;
+};
+
+export const getFilteredPeople = (
+  people: Person[],
+  { sex, query, centuries, sort, order }: Filters,
+) => {
+  let filteredPeople = [...people];
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter(person => person.sex === sex);
+  }
+
+  const normalizedQuery = query.trim().toLowerCase();
+
+  if (normalizedQuery) {
+    filteredPeople = filteredPeople.filter(person => {
+      const normalizedName = person.name.toLowerCase();
+      const normalizedMotherName = person.motherName?.toLowerCase();
+      const normalizedFatherName = person.fatherName?.toLowerCase();
+
+      return (
+        normalizedName.includes(normalizedQuery) ||
+        normalizedMotherName?.includes(normalizedQuery) ||
+        normalizedFatherName?.includes(normalizedQuery)
+      );
+    });
+  }
+
+  if (centuries.length !== 0) {
+    filteredPeople = filteredPeople.filter(person =>
+      centuries.includes(String(Math.ceil(person.born / 100))),
+    );
+  }
+
+  if (sort) {
+    filteredPeople.sort((currentPerson, nextPerson) => {
+      const currentValue = currentPerson[sort as keyof Person];
+      const nextValue = nextPerson[sort as keyof Person];
```

Comment:
The sorting logic assumes that the `sort` field will always be present in the `Person` type. Ensure that the `sort` field is a valid key of `Person` to avoid runtime errors. Consider adding a check or a default value to handle cases where the field might not exist.